### PR TITLE
feat: add analytics

### PR DESCRIPTION
**Pull request by ivanahuckova**
This PR adds simple analytics for interactive elements - buttons, inputs.

I have created `reportAppInteraction = (page: 'service_selection' | 'service_details', action: string, properties?: Record<string, unknown>) => { reportInteraction(createInteractionName(page, action), properties);};` so we are aligned with events naming and also it makes it easier to test in the future (just adding console.log above `reportInteraction`). 

Fixes: https://github.com/grafana/explore-logs/issues/256

### CODE REVIEW

File: src/services/analytics.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { reportInteraction } from '@grafana/runtime';
+import pluginJson from '../plugin.json';
+
+// Helper function to create a unique interaction name for analytics
+const createInteractionName = (page: 'service_selection' | 'service_details', action: string) => {
+  return `grafana_${pluginJson.id}_${page}_${action}`;
```
Comment (matyax) [inline src/services/analytics.ts:6]: <3

Comment (matyax) [reply src/services/analytics.ts:6]: We should consider standarizing this across observability apps. Not sure how, but we LMT could benefit from it.

File: src/Components/ServiceScene/Breakdowns/FilterByPatternsButton.tsx [hunk 2]
```diff
@@ -17,15 +18,22 @@ export class FilterByPatternsButton extends SceneObjectBase<FilterByPatternsButt
...
-    }
+    const { patterns = [] } = logExploration.state;
+
+    // Remove the pattern if it's already there
+    const filteredPatterns = patterns.filter((pattern) => pattern.pattern !== this.state.pattern);
+
+    // Analytics
+    const includePatternsLength = filteredPatterns.filter((p) => p.type === 'include')?.length ?? 0;
+    const excludePatternsLength = filteredPatterns.filter((p) => p.type === 'exclude')?.length ?? 0;
+    reportAppInteraction('service_selection', 'patterns_filtered', {
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/FilterByPatternsButton.tsx:29]: Brainstorming: we could have some JSON somewhere where we could categorize and grab these event names, potentially with some explanation. For example:

```ts
const USER_EVENTS = {
  categories: {
    // Index Page
    service_selection: 'service_selection',
  },
  subcategories: {
    service_selection: {
      // Applying pattern filter
      patterns_filtered: 'patterns_filtered',
    }
   // ...
  }
}
```

```suggestion
    reportAppInteraction(USER_EVENTS.categories.service_selection, USER_EVENTS.subcategories.service_selection.patterns_filtered, {
```

With something like this, it should be very simple to know what's being instrumented, and create dashboards without having to go to each location to get the event names.

**Review by matyax — APPROVED** [review]
🥇

File: src/Components/ServiceScene/Breakdowns/FilterByPatternsButton.tsx (hunk 2 above)
Comment (ivanahuckova) [reply src/Components/ServiceScene/Breakdowns/FilterByPatternsButton.tsx:29]: Great idea! I'll do that and re-request your review.

File: src/services/analytics.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  reportInteraction(createInteractionName(page, action), properties);
+};
+
+export const USER_EVENTS_PAGES = {
+  service_selection: 'service_selection',
+  service_details: 'service_details',
+} as const;
+
+type UserEventPagesType = (typeof USER_EVENTS_PAGES)[keyof typeof USER_EVENTS_PAGES];
+
```
Comment (matyax) [inline src/services/analytics.ts:26]: This is amazing ☄️

File: src/services/analytics.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import pluginJson from '../plugin.json';
+
+// Helper function to create a unique interaction name for analytics
+// TODO: Type action parameter
+const createInteractionName = (page: UserEventPagesType, action: string) => {
+  return `grafana_${pluginJson.id}_${page}_${action}`;
+};
+
+// Runs reportInteraction with a standardized interaction name
+// TODO: Type action parameter
```
Comment (matyax) [inline src/services/analytics.ts:11]: Wha's this todo for?

**Review by matyax — APPROVED** [review]
👏 👏

File: src/services/analytics.ts (hunk 4 above)
Comment (ivanahuckova) [reply src/services/analytics.ts:11]: Oh, I would like to type `action` parameter, so only `USER_EVENTS_ACTIONS[page]...` gets accepted. But I wasn't able to make it work and it is not a priority. I'll update it to make it more clear - what I meant.

**Pull request merged**